### ISSUE TITLE
[Bug]: When sending the max amount in native currency selecting another destination token, users encounter a warning "Insufficient funds for gas," even though the account possesses sufficient funds

### ISSUE THREAD

**Issue by sleepytanya** [MetaMask/metamask-extension#25285]
### Describe the bug

When attempting to send the maximum amount of native tokens from an account selecting another destination token, users encounter a warning on the confirmation screen indicating "Insufficient funds for gas," even though the account possesses sufficient funds.

### Expected behavior

All fees and amounts accurately calculated and displayed on the confirmation screen.

### Screenshots/Recordings

<img width="413" alt="Screenshot 2024-06-13 at 00 38 26" src="https://github.com/MetaMask/metamask-extension/assets/104780023/5ec0907c-934c-465e-aafb-0a9142c25193">

https://github.com/MetaMask/metamask-extension/assets/104780023/25eeb236-c28b-45db-affa-961749625cf8

### Steps to reproduce

1. Start send+swap transaction
2. Select native token as an origin and another token for the destination
3. Set Max amount
4. See the warning about insufficient funds

### Version

12.0.0

### Build type

None

### Browser

Chrome, Firefox

### Operating system

MacOS

**Comment by BZahory** [comment]
I don't believe this is a bug. The `insufficient funds for gas` warning is shown based on the swap+send gas, not the basic send gas used for the "max" calculation. With that, the UI here could better communicate that so I'm leaving this open for now, but I don't think this would be sev2

**Comment by danjm** [comment]
@sleepytanya If you take the same action with the same account on the current production build (v11.16.12) do you get the same warning? Or are you able to proceed with the transaction?

**Comment by sleepytanya** [comment]
@danjm v11.16.12 doesn't have send+swap, just simple Send flow. Using Max on v11.16.12 doesn't cause any issues. 

> Or are you able to proceed with the transaction?
   On 12.0.0 the transaction can't be confirmed, the 'Confirm' button is disabled. If user selects non-native<->non-native token for send+swap then 'Max' amount is calculated correctly:

<img width="363" alt="Screenshot 2024-06-19 at 23 53 21" src="https://github.com/MetaMask/metamask-extension/assets/104780023/fa3b30b3-3e3d-4088-9d63-6efa5063e6af">

<img width="362" alt="Screenshot 2024-06-19 at 23 53 53" src="https://github.com/MetaMask/metamask-extension/assets/104780023/0dd2ad2f-eaa5-4807-8c8c-371e4075871c">

**Comment by hesterbruikman** [comment]
@BZahory @sleepytanya can we define expected behavior as follows:

GIVEN Native to non-native
AND Token amount is Max
THEN No funds remain for gas
>> Inform the user that max amount cannot be used and recommend to reduce token amount
(copy to be worked on with content design)

**Comment by ejwessel** [comment]
| srcToken | destToken | insufficient funds? |
| --- | --- | --- |
| nativeToken | nativeToken | true |
| nativeToken | nonNative | true |
| nonNative | nonNative | false |
| nonNative | nonNative | false |

**Comment by sleepytanya** [comment]
Latest v12.0.0 build https://github.com/MetaMask/metamask-extension/pull/25098#issuecomment-2251300111


https://github.com/user-attachments/assets/c68a6531-ce70-4bc0-b29f-7cb484c6cbe3

**Comment by ejwessel** [comment]
@sleepytanya can we close this out as it looks like it was merged in release 12.0?

**Comment by sleepytanya** [comment]
@ejwessel Closing!

nativeToken -> nativeToken
nativeToken -> ERC20 (Max option is unavailable)
ERC20         -> nativeToken
ERC20         -> ERC20

All scenarios above work as expected in 12.0.0.




https://github.com/user-attachments/assets/25c843d5-a129-4e61-bfdc-6f1e4efb4d4e
